mantidproject/mantid
# Fix bug in CentroidPeaks with EdgePixels option

Description of work.
Check that results of CentroidPeaks are inside panels.  Fix bug in PeaksWorkspace::removePeaks.

**To test:**
```python
Load(Filename='TOPAZ_3132_event.nxs', OutputWorkspace='TOPAZ_3132_event')
FilterBadPulses(InputWorkspace='TOPAZ_3132_event', OutputWorkspace='TOPAZ_3132_event', LowerCutoff=25)
ConvertToMD(InputWorkspace='TOPAZ_3132_event', QDimensions='Q3D', dEAnalysisMode='Elastic', Q3DFrames='Q_sample', LorentzCorrection=True, OutputWorkspace='TOPAZ_3132_md', MinValues='-25,-25,-25', MaxValues='25,25,25', SplitInto='2', SplitThreshold=50, MaxRecursionDepth=13, MinRecursionDepth=7)
FindPeaksMD(InputWorkspace='TOPAZ_3132_md', PeakDistanceThreshold=0.37680000000000002, MaxPeaks=50, DensityThresholdFactor=100, OutputWorkspace='TOPAZ_3132_peaks')
CentroidPeaks(InPeaksWorkspace='TOPAZ_3132_peaks', InputWorkspace='TOPAZ_3132_event', PeakRadius=4, EdgePixels=20, OutPeaksWorkspace='TOPAZ_3132_peaks')
```
Then open TOPAZ_3132_peaks in MantidPlot.  Should see this error before fix and not after:
![centroidpeaks](https://user-images.githubusercontent.com/1108354/29893259-7715f4e0-8d9f-11e7-8e39-cafdd64ab6b2.png)


Fixes #20355 

**Release Notes** 

*Does not need to be in the release notes since fixing bug introduced in this release.*


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
